src/ui/pages/work/partusage/PartUsageDialog.js

```diff
@@ -129,71 +143,94 @@ function PartUsageDialog(props) {
+        setIsTrackedByAsset(false);
     };
 
     const handleTransactionChange = () => {
-        resetFormTransactionLinesAndBinListStates();
+        resetFormTransactionLinesAndRelatedStates();
     };
 
     const handleStoreChange = () => {
-        resetFormTransactionLinesAndBinListStates();
+        resetFormTransactionLinesAndRelatedStates();
     };
 
     /* This function handles at least 3 cases:
-     * 1) The asset ID input is cleared.
-     * 2) The user searches for an asset ID directly (without having selected a part),
-     *    so we have to update the remaining state concerning part and bin.
-     * 3) There was already a part selected so we only need to update the bin state.
+     * 1) The asset ID field is cleared, is solely whitespace or is falsy.
+     * 2) The user searches for an asset ID without having selected a part or by selecting it from the history
+     *    (so we need to update the related state: part, bin and lot).
+     * 3) There was already a part selected (so we only need to update the bin and lot states).
      */
     const handleAssetChange = async (assetIDCode) => {
-        // Asset input cleared
-        if (assetIDCode?.trim() === '') {
-            updateFormDataProperty(FORM.ASSET, '');
-            updateFormDataProperty(FORM.ASSET_DESC, '');
-            updateFormDataProperty(FORM.BIN, '');
-            setBinList([]);
+        const { transactionType, partCode } = formData;
+
+        // Reset related fields
+        updateFormDataProperty(FORM.BIN, '');
+        if (transactionType === issueType) {
+            // In a return transaction the bin list is loaded when a part is selected so no need to reset
+            resetFieldWithList(FORM.BIN, setBinList);
+        }
+        resetFieldWithList(FORM.LOT, setLotList);
+
+        // Asset field is definitely not an asset code
+        if (!assetIDCode || assetIDCode?.trim() === '') {
+            // Reset state that may not reflect the value shown in the field
+            resetFieldWithDesc(FORM.ASSET, FORM.ASSET_DESC);
             return;
         }
+
         try {
-            setLoading(true);
             const assetData = await getAssetData(assetIDCode);
+
+            // Explicitly reset asset field since the selected asset was not valid
             if (!assetData) {
-                updateFormDataProperty(FORM.ASSET, '');
-                updateFormDataProperty(FORM.ASSET_DESC, '');
+                resetFieldWithDesc(FORM.ASSET, FORM.ASSET_DESC);
                 return;
             }
-            // Asset selected without a part selected
-            if (!formData.partCode || formData.partCode && formData.partCode !== assetData.partCode) {
-                return handleAssetUnspecifiedPart(assetData);
-            }
+
+            // Asset selected without a part selected or with a different part associated (if selected from history)
+            if (!partCode || partCode && partCode !== assetData.partCode) {
```

Comment:
Why are we checking partCode is not falsy again after the left operand checks it?

Reply:
not needed indeed